cpp_utils/src/cpp/math/random/TSafeRandomManager.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file TSafeRandomManager.cpp
+ *
+ */
+
+#include <cstdlib>
+#include <mutex>
+
+#include <cpp_utils/math/random/TSafeRandomManager.hpp>
+
+namespace eprosima {
+namespace utils {
+
+TSafeRandomManager::TSafeRandomManager(const RandomSeedType& original_seed /* = 1 */ )
+    : std_random_generator_(original_seed)
+{
+    // Do nothing
+}
+
+RandomNumberType TSafeRandomManager::pure_rand () noexcept
+{
+    return pure_random_generator_();
```

Comment:
Is this thread safe?

Reply:
It is, even when it is not documented :sweat_smile: 
It does not have an inner state, so it is.